Packages/com.unity.inputsystem/InputSystem/Plugins/HID/HID.cs

```diff
@@ -319,16 +326,88 @@ public static bool UsageToString(UsagePage usagePage, int usage, out string usag
+                        bitOffset = yElement.reportOffsetInBits % 8;
+                        byteOffset = yElement.reportOffsetInBits / 8;
+                        sizeInBits = xElement.reportOffsetInBits - yElement.reportOffsetInBits + xElement.reportSizeInBits;
+                    }
+
+                    var stickName = "Stick";
+                    var control = builder.AddControl(stickName)
+                        .WithLayout("Stick")
+                        .WithBitOffset((uint)bitOffset)
+                        .WithByteOffset((uint)byteOffset)
+                        .WithSizeInBits((uint)sizeInBits)
+                        .WithUsages(new InternedString[] { CommonUsages.Primary2DMotion });
+
+                    builder.AddControl(stickName + "/x")
+                        .WithFormat(InputStateBlock.kTypeSBit)
+                        .WithLayout("Axis")
+                        .WithBitOffset(0)
+                        .WithSizeInBits((uint)xElement.reportSizeInBits);
+
+                    builder.AddControl(stickName + "/y")
+                        .WithFormat(InputStateBlock.kTypeSBit)
+                        .WithLayout("Axis")
+                        .WithBitOffset((uint)(yElement.reportOffsetInBits - xElement.reportOffsetInBits))
+                        .WithSizeInBits((uint)xElement.reportSizeInBits);
+
+                    //Need to handle Up/Down/Left/Right
+                    builder.AddControl(stickName + "/up")
+                        .WithFormat(InputStateBlock.kTypeSBit)
+                        .WithLayout("Button")
+                        .WithParameters("clampMin=0,clampMax=1")
+                        .WithBitOffset((uint)(yElement.reportOffsetInBits - xElement.reportOffsetInBits))
+                        .WithSizeInBits((uint)xElement.reportSizeInBits);
+
+                    builder.AddControl(stickName + "/down")
+                        .WithFormat(InputStateBlock.kTypeSBit)
+                        .WithLayout("Button")
+                        .WithParameters("clamp,clampMin=-1,clampMax=0,invert")
+                        .WithBitOffset((uint)(yElement.reportOffsetInBits - xElement.reportOffsetInBits))
+                        .WithSizeInBits((uint)xElement.reportSizeInBits);
+
+                    builder.AddControl(stickName + "/left")
+                        .WithFormat(InputStateBlock.kTypeSBit)
+                        .WithLayout("Button")
+                        .WithParameters("clamp,clampMin=-1,clampMax=0,invert")
+                        .WithBitOffset(0)
+                        .WithSizeInBits((uint)xElement.reportSizeInBits);
+
+                    builder.AddControl(stickName + "/right")
+                        .WithFormat(InputStateBlock.kTypeSBit)
+                        .WithLayout("Button")
+                        .WithParameters("clampMin=0,clampMax=1")
+                        .WithBitOffset(0)
+                        .WithSizeInBits((uint)xElement.reportSizeInBits);
```

Comment:
Are left/right/up/down actually needed here? The default setup in stick seems like it should do the trick. It's wired to use the state from X and Y and should thus automatically get the format, offsets, and sizes and such (assuming there's no bug there, of course :)).

Reply:
It was not getting the proper state.  It was expecting 32 bits for each X and Y, and working off that.  Which meant it was overreadding.

Reply:
Crap, means that there's a problem in the `useStateFrom` logic. Probably something with it not working in combination with another layout retroactively changing the state layout. I'll add a ////TODO.